### PR TITLE
fix: retry on mergeable state unknown

### LINKED ISSUE #674: Retry on mergeableState: UNKNOWN
**How would you describe the issue?**

We have a bot that updates dependencies once a week and it creates PRs in quick succession.
Once the first PR is merged automatically, other PRs see `mergeableState: UNKNOWN` for a short while.
Sometimes you can even observe that in the GitHub UI (the merge button is greyed out).
We tried increasing the `MAXIMUM_RETRIES`, but it looks like the action is not retrying in this case.

**How can we reproduce the issue?**

I'm not sure if it can be reproduced manually without a bot.
But basically create multiple PRs in quick succession.
Here is an example run: https://github.com/moia-dev/scynamo/runs/1548816898#step:2:7

**What are the expected results?**

Retry on `mergeableState: UNKNOWN` according to the `MAXIMUM_RETRIES` setting.

**What are the actual results?**

AFAICT there were no retries in that run.

**How much does it hurt?**

Unfortunately this problem breaks our use case.

### PR DESCRIPTION
**Pull request by xtazz**
This Pull Request fulfills the following requirements:

- [X] The commit message follows our guidelines.
- [X] Tests for the changes have been added if needed.
- [x] Does not affect documentation or it has been added or updated.

Resolves #674

### CODE REVIEW

File: src/eventHandlers/continuousIntegrationEnd/index.spec.ts [hunk 1]
```diff
@@ -310,6 +310,82 @@ describe('continuous integration end event handler', (): void => {
     );
   });
 
+  it('retries fetching pull request information for which mergeable status is unknown', async (): Promise<void> => {
+    expect.assertions(1);
+
+    const firstResponse: Response = {
```
Comment (acazacu) [inline src/eventHandlers/continuousIntegrationEnd/index.spec.ts:316]: Maybe not the scope of this PR, but having a generator for the response would make some of this code shorter.

**Review by viestat — APPROVED** [review]

File: src/common/delay.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const EXPONENTIAL_BACKOFF = 2;
+export const MINIMUM_WAIT_TIME = 1000;
+
+export const delay = async (duration: number): Promise<void> =>
```
[the remark this answers is not in the document]
Comment (xtazz) [reply src/common/delay.ts:4]: Yes

File: src/eventHandlers/continuousIntegrationEnd/index.spec.ts (hunk 1 above)
Comment (xtazz) [reply src/eventHandlers/continuousIntegrationEnd/index.spec.ts:316]: Let's do it separately, it should be updated for all tests like this one

Comment (xtazz) [reply src/eventHandlers/continuousIntegrationEnd/index.spec.ts:316]: Created a task for it

**Pull request merged**